src/Setup_geometry.jl

```diff
@@ -731,6 +754,97 @@ function add_polygon!(Phase, Temp, Grid::AbstractGeneralGrid;   # required input
     return nothing
 end
 
+"""
+        add_plate!(Phase, Temp, Grid::AbstractGeneralGrid; xlim=(), ylim=(), zlim::Tuple = (0.0,0.8), phase = ConstantPhase(1), T=nothing, segments=nothing, cell=false )
+
+Adds a tectonic plate with phase and temperature structure to a 3D model setup.
+This function enables the definition of tectonic plates in the xy plane and projects them along the z-axis, providing a flexible approach to model complex plate geometries.
+
+Parameters
+==========
+- `Phase`  - Phase array (consistent with Grid)
+- `Temp`   - Temperature array (consistent with Grid)
+- `Grid`   - Grid structure (usually obtained with read_LaMEM_inputfile)
+- `xlim`   - `x`-coordinate of the polygon points, same ordering as ylim, number of points unlimited
+- `ylim`   - `y`-coordinate of the polygon points, same ordering as xlim, number of points unlimited
+- `zlim`   - `z`-coordinate range for projecting the polygon (start and stop, two values)
+- `phase`  - Specifies the phase of the plate. See `ConstantPhase()`
+- `T`      - Specifies the temperature of the plate. See `ConstantTemp()`, `LinearTemp()`, `HalfspaceCoolingTemp()`, `SpreadingRateTemp()`
+- `segments` - Optional. Allows for thermal segmentation within the polygon. Useful for ridge systems or complex thermal structures.
+- `cell`   - If true, `Phase` and `Temp` are defined on cell centers
+
+Example
+========
+
+Tectonic plate in the xy plane with phase and temperature structure:
+
+```julia-repl
+julia> Grid = CartData(xyz_grid(x, y, z))
+
+Grid:
+  nel         : (512, 512, 128)
+  marker/cell : (1, 1, 1)
+  markers     : (512, 512, 128)
+  x           ϵ [-1000.0 : 0.0]
+  y           ϵ [-1000.0 : 1000.0]
+  z           ϵ [-660.0 : 0.0]
+julia> Phases = zeros(Int32,   size(Grid.X))
+julia> Temp   = zeros(Float64, size(Grid.X))
+julia> segments = [
+           ((-500.0, -1000.0), (-500.0, 0.0)),  # Segment 1
+           ((-250.0, 0.0), (-250.0, 200.0)),    # Segment 2
+           ((-750.0, 200.0), (-750.0, 1000.0))  # Segment 3
+       ]
+julia> lith = LithosphericPhases(Layers=[15 55], Phases=[1 2], Tlab=1250)
+julia> add_plate!(Phases, Temp, Grid; 
+           xlim=(-1000.0, -750.0, -250.0, 0.0, -250.0, -750.0), 
+           ylim=(0.0, 500.0, 500.0, 0.0, -500.0, -500.0), 
+           zlim=(-150.0, 0.0), 
+           phase=lith, 
+           T=SpreadingRateTemp(SpreadingVel=3), 
+           segments=segments)
+julia> Grid = addfield(Grid, (; Phases, Temp))  # Add fields
+julia> write_paraview(Grid, "Plate")  # Save model to Paraview
+1-element Vector{String}:
+ "Plate.vts"
+"""
+
+function add_plate!(Phase, Temp, Grid::AbstractGeneralGrid;   
+    xlim=(), ylim=(), zlim::Tuple = (0.0,0.8),           
+    phase = ConstantPhase(1),                                   
+    T=nothing, segments=nothing, cell=false )                            
+
+    xlim_ = Float64.(collect(xlim))
+    ylim_ = Float64.(collect(ylim))
+    zlim_ = Float64.(collect(zlim))
```

Comment:
```suggestion
    xlim_ = collect(xlim)
    ylim_ = collect(ylim)
    zlim_ = collect(zlim)
```
Is the `Float64` conversion really needed for any reason? otherwise it's better to be generic.

Reply:
Thank you. I will change it to be generic.